### PR TITLE
Adding float16 arithmetic

### PR DESCRIPTION
**Pull request by ssk1328**
This adds modules for implementing floating point multiplication and addition. Tested to work in simulation and in hardware.

### CODE REVIEW

**Comment by mithro** [comment]
Please put WIP in the pull request title if this is a WIP.

File: litevideo/float_arithmetic/common.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+            o.eq(maximum)
+        ).Elif(i < minimum,
+            o.eq(minimum)
+        ).Else(
+            o.eq(i)
+        )
+    ]
+
+def in_layout(dw):
+    return [("a", dw), ("b", dw)]
```
Comment (mithro) [inline litevideo/float_arithmetic/common.py]: a b?

File: litevideo/float_arithmetic/common.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+from litex.gen import *
+
+from litex.soc.interconnect import stream
+
+
+def saturate(i, o, minimum, maximum):
```
Comment (mithro) [inline litevideo/float_arithmetic/common.py]: Add docstring

File: litevideo/float_arithmetic/floatadd.py [hunk 3]
```diff
@@ -0,0 +1,251 @@
+# floatadd
```
Comment (mithro) [inline litevideo/float_arithmetic/floatadd.py]: Add module level docstring.

**Comment by mithro** [comment]
I don't think the `lena.png` file is actually used in this code anywhere?

**Comment by ssk1328** [comment]
Current Status:

- Both mult and float units tested to be working in simulation. Simulated in the HDMI2USB-misoc-firmware litex environment

- Can't figure out the reason for compilation error when ported to litex, more details in description

- Not tested in hardware, can't directly test on hardware in the litex version, and once tested in hardware need to port back to litex version from migen version. Lot of switching.

**Comment by ssk1328** [comment]
Status Update: 

- Tested to be working in hardware in the HDMI2USB Migen environment, using CSR registers.

**Comment by mithro** [comment]
@enjoy-digital Can we also merge this change too?

**Pull request merged**